### PR TITLE
Add security guide for web-based projects

### PR DESCRIPTION
**Pull request by zallarak**
Simple checks could have prevented the MyEtherWallet DNS hack, and the BlackWallet client-side script injection; this guide covers them and more.

As more projects are released, we should publish basic guidelines that outline basic web-sec standards for people building Stellar apps. We shouldn't be the authority on this, but we should cover important concepts for Stellar-app developers to ensure our ecosystem stays healthy.

### CODE REVIEW

**Comment by theaeolianmachine** [comment]
@stanford-scs, would you mind giving a look over this doc - is there anything else we'd like to have covered? Happy to go over in person.

**Comment by theaeolianmachine** [comment]
Thanks for sending this over @zallarak!

**Comment by zallarak** [comment]
Just a friendly bump on this, cc @theaeolianmachine.

**Comment by zallarak** [comment]
@stanford-scs thanks for your **very** valuable feedback! Will make an iteration that addresses your comments.

**Comment by tomquisel** [comment]
@zallarak We're looking forward to your iteration!

**Comment by theaeolianmachine** [comment]
Hey @zallarak just checking in here!

**Comment by zallarak** [comment]
Sorry for the delay! I'm back on this. I updated per comments.

@stanford-scs I incorporated all of the suggested edits. Regarding the question-list you left; I think many of those issues have many ways of being solved. However, my intent in making this guide was to take projects from zero security to a baseline that is "acceptable", since anyone can technically build on Stellar. I think it still adds value in its current form without getting too specific. I am open to adding deep-dives separately for the wallet issues you listed; does that sound OK to you?

Again, pardon my delay, promise to be quicker on this.

**Comment by ire-and-curses** [comment]
+1 for getting this out sooner and then doing something deeper. This is already a *ton* better than the present situation (nada).

File: guides/walkthroughs/securing-web-projects.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+Finally, it is worth setting up errror reporting (e.g. [Sentry](https://sentry.io/welcome/)): Oftentimes, people trigger strange bugs when they are trying to hack things.
+
+# Authentication weaknesses
+
+If you have logins for users, it is critical that your authentication system is built securely. Of course, the best way to do this is to use something off the shelf. Both Ruby on Rails and Django have robust, built-in authentication schemes.
+
+Many JSON web token implementations are poorly done, so ensure the library you use is audited.
+
+Hash passwords with a time-tested scheme. The winner of the last password hashing contest was Argon2. Baloon Hashing is also worth looking into.
```
Comment (theaeolianmachine) [inline guides/walkthroughs/securing-web-projects.md]:
```suggestion
Hash passwords with a time-tested scheme. The winner of the last password hashing contest was Argon2. Balloon Hashing is also worth looking into.
```

**Comment by zallarak** [comment]
@theaeolianmachine great catch. I also reformatted to 100 char width - good call.

**Comment by theaeolianmachine** [comment]
Thanks again @zallarak!

**Pull request merged**